include/rocksdb/types.h

```diff
@@ -21,6 +21,7 @@ const SequenceNumber kMinUnCommittedSeq = 1;  // 0 is always committed
 enum EntryType {
   kEntryPut,
   kEntryDelete,
+  kEntryDeleteWithTimestamp,
```

Comment:
enum order changes kind of scare me. For example, someone might store or transfer some keys/values along with the EntryType. In that case they likely encode EntryType as an int. So upgrading, or transferring to a host running an older/newer version will decode that int as a different EntryType from what was intended.

We could comment on this enum that the ordering is subject to change.

Reply:
It scares me too. Imo, int-based enums should always have explicit value assignments. I looked at previous PRs changing this enum, and they also changed the order...

Reply:
okay, let me not change the ordering of enum.